api/dps/index.go

```diff
@@ -195,15 +195,15 @@ func (i *Index) Transactions(blockID flow.Identifier) ([]flow.Identifier, error)
 	}
 	res, err := i.client.ListTransactionsForBlock(context.Background(), &req)
 	if err != nil {
-		return nil, fmt.Errorf("could not get transaction: %w", err)
+		return nil, fmt.Errorf("could not get transactions: %w", err)
 	}
 
 	var transactions []flow.Identifier
 	for _, id := range res.TransactionIDs {
 		var transactionID flow.Identifier
 		err = cbor.Unmarshal(id, &transactionID)
 		if err != nil {
-			return nil, fmt.Errorf("could not decode transaction ID: %w", err)
+			return nil, fmt.Errorf("could not decode transaction ID %q: %w", id, err)
```

Comment:
I would usually use `%x` here.

Generally, I like the approach of putting parameters in brackets before the error reason:

```go
return nil, fmt.Errorf("could not do action (param1: %x, param2: %s): %w", param1, param2, err)
```